### PR TITLE
fix: enable celery autoscaling

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR removes the [concurrency](https://docs.celeryq.dev/en/stable/userguide/workers.html#concurrency) parameter of celery in favor of [autoscale](https://docs.celeryq.dev/en/stable/userguide/workers.html#autoscaling) which provides a minimum and a maximum concurrency, allowing Celery to scale based on the workload.

A fixed number of `concurrency` is not optimal for any workload.

### Author concerns

By default, Celery uses the prefork pool which uses Python's multiprocessing package which uses only the available CPU cores so, setting the max limit of `autoscale` to 10 can be problematic. However, this would make a more efficient use of available hardware and allow any HPA resource to scale better.

### CODE REVIEW

**Comment by MoisesGSalas** [comment]
IIRC we had an issue with long running tasks and node scaling that caused an increase in tasks failed. Can you double check that?

> By default, Celery uses the prefork pool which uses Python's multiprocessing package which uses only the available CPU cores so, setting the max limit of autoscale to 10 can be problematic. However, this would make a more efficient use of available hardware and allow any HPA resource to scale better.


Do the values of `autoscale` modify the same thing as [`worker_concurrency`](https://docs.celeryq.dev/en/stable/userguide/configuration.html#worker-concurrency)? In that case isn't it better to let celery use the default value for `worker_concurrency` (the same number of cores as de machine), given that the recommendation is to use the same number unless your tasks are heavy on I/O?

Ideally the HPA should scale on number of tasks, but we are not there yet.

Also, try to link the documentation for each flag.

**Comment by Ian2012** [comment]
Yes, it's the same. We should measure the performance gains of disabling the autoscale component.

**Comment by jfavellar90** [comment]
@Ian2012 according to your testing, we should not include the autoscaling parameter. I agree on the value of 100 for `max-tasks-per-child`.

**Comment by Ian2012** [comment]
@jfavellar90 The configuration would be the same as the default tutor deployment: https://github.com/overhangio/tutor/blob/v17.0.2/tutor/templates/k8s/deployments.yml#L144 so I don't see why the celery tuning plugin should exist. Unless we use it to manage different queues and deployments as proposed.

File: drydock/templates/drydock/k8s/celery/lms-worker.yml [hunk 1]
```diff
@@ -13,8 +13,8 @@ spec:
             - --app=lms.celery
             - worker
             - --loglevel=info
-            - --concurrency=1
+            - --autoscale=10,1
```
Comment (jfavellar90) [inline drydock/templates/drydock/k8s/celery/lms-worker.yml:16]: @Ian2012 as we agreed, we should remove the `autoscale` parameter

Comment (MoisesGSalas) [reply drydock/templates/drydock/k8s/celery/lms-worker.yml:16]: I think is better to close the PR at the moment and revisit celery optimizations later.

As @Ian2012 said, removing the `autoscale` flag leaves the same command invocation as vanilla tutor. In that case is just easier to simply disable the plugin by default. Or delete the whole plugin (although maybe there's a situation on which the original plugin behavior is preferred?)

**Pull request closed without merging**